tests/unit/meta/test_system_user.py

```diff
@@ -71,3 +71,73 @@ def test_invalid_user_non_shared_scope_from_dict(self):
             error.get_brief(),
             Equals("Invalid system-usernames for 'name': scope 'ghost' is invalid"),
         )
+
+    def test_from_object_dict(self):
+        user_name = "name"
+        user_object = {"scope": "shared"}
+
+        user = SystemUser.from_object(user_name=user_name, user_object=user_object)
+
+        self.assertThat(user.name, Equals(user_name))
+        self.assertThat(user.scope, Equals(SystemUserScope.SHARED))
+
+    def test_from_object_dict_invalid(self):
+        user_name = "name"
+        user_object = {"scope": "invalid_scope"}
+
+        error = self.assertRaises(
+            errors.SystemUsernamesValidationError,
+            SystemUser.from_object,
+            user_name=user_name,
+            user_object=user_object,
+        )
+
+        self.assertThat(
+            error.get_brief(),
+            Equals(
+                "Invalid system-usernames for 'name': scope 'invalid_scope' is invalid"
```

Comment:
let's please keep message string comparisons on test_errors

Reply:
I can do that, though test_errors really just ensures the error class is correct.

The assertRaises checks the correct type.

The string check ensures correct usage of the init parameters, though that could be done on the object fields too (i.e. check error._name, error._message). Or just skip altogether?

Reply:
nvm, see "check for attributes" below. I concur

Reply:
done